### PR TITLE
Release 0.7

### PR DESCRIPTION
**Pull request by krl**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,27 +7,133 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Added the `MappedBranch` to provide branches that only allow access to certain parts of its leaves
+- Added implementation of `IntoIterator` for `Branch` and `BranchMut`
+- Added `First` auto-trait to construct a `Branch` to the first element in a collection
+- Added `LinkedList` implementation in tests
+
+### Changed
+
+- Changed `canonical`/`canonical_derive` version from 0.5 to 0.6
+- Split up the `Annotation` trait into `Annotation` for the leaves, and `Combine` for the nodes.
+- Changed the iterator on `Compound` to only iterate over populated subtrees or leaves.
```
Comment (ZER0) [inline CHANGELOG.md:26]: few nits:

1. The section is past tense ("Added", "Changed", etc) but the item list inside are present tense ("Add", "Change", etc)
2. if we can't rephrase the "Split up" into "Change", I would suggest to use "Refactor" that is already an alias for "Change"
3. Do not use the period at the end of the list item.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,27 +7,133 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Removed
+
+- Removed the `Annotation` trait parameter on `Compound` moving to a generic on the type.
```
Comment (ZER0) [inline CHANGELOG.md:30]: Same here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,27 +7,133 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [0.5.5] - 03-12-20
+### Added
```
Comment (ZER0) [inline CHANGELOG.md:10]: If you want to merge into the default branch it means those changes belongs to 0.7.0, not Unreleased.

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,27 +7,133 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Added
+
+- Add CI infrastructure
+- Add branch introspection via `levels` method
+
+### Changed
+
+- Changed `Branch::len` to `Branch::depth`
+- Changed the library to be no_std compatible
```
Comment (ZER0) [inline CHANGELOG.md:103]: Nit: use present tense

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,27 +7,133 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Add LICENSE and copyright notices
+- Add `Nth` trait for trees
+- Add capacity to search through trees by walking
+- Add `Branch` and `BranchMut`
+
+## [0.1.0] - 2020-10-16
+
+Initial
+
+[Unreleased]: https://github.com/dusk-network/microkelvin/compare/canonical-0.6.0...HEAD
```
Comment (ZER0) [inline CHANGELOG.md:125]: Nit: wrong tag used (`canonical-0.6.0`)

**Review by ZER0 — CHANGES_REQUESTED** [review]
Few nits to fix.

Also about the commit message, remember to follow this: https://chris.beams.io/posts/git-commit/

And in our case also to start the summary with one of the CHANGELOG "verbs" ("Add", "Remove", "Change", "Fix", etc. Consider "Replace", "Refactor" and "Rename" as alias of "Change"); since at least ideally the would be used to generate a draft CHANGELOG automatically, and I want to be sure we're consistent already (and we get use to this).

File: CHANGELOG.md (hunk 1 above)
Comment (krl) [reply CHANGELOG.md]: got it

**Pull request merged**